src/main/java/uk/gov/hmcts/cft/idam/api/v2/common/auth/IdamClientCredentialsConfig.java

```diff
@@ -0,0 +1,40 @@
+package uk.gov.hmcts.cft.idam.api.v2.common.auth;
+
+import feign.RequestInterceptor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.security.oauth2.client.AuthorizedClientServiceOAuth2AuthorizedClientManager;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientManager;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientProviderBuilder;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientService;
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+
+public class IdamClientCredentialsConfig {
+
+    @Value("${idam.client.registration.id}")
```

Comment:
Do we need a separate property - can we use the client_id of the registered OAuth2 client?

Reply:
Riiight, I follow you now!